File: Radio/AudioCaptureServer/DataBase/Models/Audio.cs
using Microsoft.EntityFrameworkCore;

namespace AudioCaptureServer.DataBase.Models;

[PrimaryKey("fileName")]
public class Audio
{
    public string? radioName { get; init; }
    public string fileName { get; init; }
    public string folderName { get; init; }
    public DateTime startRecording { get; init; }
    public DateTime endRecording { get; init; }
    
    public int status { get; init; }

    public Audio(string? radioName, string fileName, string folderName, DateTime startRecording, DateTime endRecording,int status)
    {
        this.radioName = radioName;
        this.fileName = fileName;
        this.folderName = folderName;
        this.startRecording = startRecording;
        this.endRecording = endRecording;
        this.status = status;
    }
}

File: Application/lib/DataClasses/Audio.dart
class MyAudio{
  String? radioName;
  String fileName;
  String folderName;
  DateTime startRecording;
  DateTime endRecording;
  int status;

  MyAudio({required this.radioName,required this.fileName,required this.folderName,required this.startRecording,required this.endRecording,required this.status});

  factory MyAudio.fromJson(Map<String, dynamic> json) {
    return MyAudio(
      radioName: json['radioName'],
      fileName: json['fileName'],
      folderName: json['folderName'],
      startRecording: DateTime.parse(json['startRecording']),
      endRecording: DateTime.parse(json['endRecording']),
      status: json['status']
    );
  }
}